pallets/poa/src/lib.rs

```diff
@@ -283,10 +283,11 @@ decl_module! {
         /// In worst case there is a write to `QueuedValidators`.
         /// In case of short circuit there is 1 read and write to `EpochEndsAt`
         /// # </weight>
-        #[weight = (T::DbWeight::get().reads_writes(2 + *short_circuit as Weight, 1 + *short_circuit as Weight), Pays::No)]
-        pub fn add_validator(origin, validator_id: T::AccountId, short_circuit: bool) -> dispatch::DispatchResult {
+        #[weight = T::DbWeight::get().reads_writes(2 + *short_circuit as Weight, 1 + *short_circuit as Weight)]
+        pub fn add_validator(origin, validator_id: T::AccountId, short_circuit: bool) -> dispatch::DispatchResultWithPostInfo {
             ensure_root(origin)?;
-            Self::add_validator_(validator_id, short_circuit)
+            Self::add_validator_(validator_id, short_circuit)?;
+            Ok(Pays::No.into())
```

Comment:
This return value has no effect when called using sudo. [sudo calls were recently changed to never pay](https://github.com/paritytech/substrate/blob/35fe3cd1bc4b64cadb0bc6196ae40173db65bb28/frame/sudo/src/lib.rs#L142).

Same is true for the other similar changes (except `migrate` which isn't called by sudo).

Reply:
This is to help when called by Master